### ISSUE TITLE
Preparse in a category

### ISSUE THREAD

**Issue by outline4** [aelvan/Preparse-Field-Craft#34]
I would like to add a preparse field to a category in order to see the related entries (in my case products) of this category. 

Does preparse work in categories or only in entries? 

Even the most basic thing like {{ element.id }} won't work. 

Thanks in advance

**Comment by aelvan** [comment]
Are you using the Craft 2 or Craft 3 version?

**Comment by outline4** [comment]
Hi André, 

Sorry I am using Craft 2. 


Cheers
Stefan

**Comment by aelvan** [comment]
Right, then you should user `category` in your templates, ie `{{ category.id }}`. 

Try that and let me know if it helps.

**Comment by outline4** [comment]
Hi André, 

That was it… Now I understand what you mean by element in the docs. 
It’s category, if it’s in a category, it’s entry when it’s in a entry etc… mI right? 

I try to plot out entries with links to those entries… But the links are encoded… It there a workaround? 

Here’s my code:
{% set products = craft.commerce.products.relatedTo(category) %}
{% if products|length %}{% for product in products %}<a href="/backend/commerce/products/product/{{ product.id }}-{{ product.slug}}">{{ product.title }}</a>, {% endfor %}{% endif %}

Any help highly appreciated! 
Stefan

**Comment by aelvan** [comment]
Yeah, in the Craft 2 version, the element you're on is accessible by the element type handle, so `category` if it's a category, `asset` if it's an Asset, and so on. In the Craft 3 version, this changed a wee bit, because custom elements doesn't necessarily have a handle anymore. So there the element is always accessible by `element`, and also the element type handle if it's a built in element type.

You probably need to output it through the raw filter if you store html, something like `{{ category.myPreparseField | raw }}`.

**Comment by outline4** [comment]
Hi andré, 

unluckily the raw thing does not work… neither other filters like encode etc. 

I’ve tried to do this:

	{% set products = craft.commerce.products.relatedTo(category) %}
	{% if products|length %}{% for product in products %}{% filter raw %}<a href="/backend/commerce/products/product/{{ product.id }}-{{ product.slug}}">{{ product.title }}</a>, {% endfilter %}{% endfor %}{% endif %}

Hm?

Cheers
Stefan

**Comment by aelvan** [comment]
What I meant was that you need to use raw when you output the value from the preparse field, not in the code that creates whatever goes into it. Is that what you tested?

Or maybe I misunderstood what you were trying to do..? Happens alot. :)

**Comment by outline4** [comment]
Hi André, 

LOL :) 
But yeah, you misunderstood me. Probably because I’ve explained it wrong. Happens a lot as well :)

In fact, I want to show these related Products in the backend! 

Something like this: 



Maybe it’s a total misuse of your plugin… That’s also possible…

As a feature request, I would love to have a html output field… that I could optionally style. 

Cheers
Stefan

**Comment by aelvan** [comment]
Ah, I see. You want to show that HTML on the entry? Yeah, that's not really what the field is for. A fieldtype doing that would actually be alot easier to make than what the Preparse Field does. ;) That twig code you put in is parsed and then the value is saved to the field as a value. It's not meant to display that information in the CP, that's mainly just for debugging purposes.

**Comment by outline4** [comment]
I see… 

Your field is context related (asset, entries, users etc.), which would make it great for backend enhancements :) 
In fact I wanted to mimic the introvert plugin behavior, but for products. 
My bad. 

Still, I couldn’t live without preparse…

Thanks anyways! 
Stefan
